libraries/botframework-connector/src/auth/openIdMetadata.ts

```diff
@@ -56,7 +59,12 @@ export class OpenIdMetadata {
      * @private
      */
     private async refreshCache(): Promise<void> {
-        const res = await fetch(this.url);
+        let agent = null;
+        if (this.proxySettings) {
+            const proxyUrl = `http://${this.proxySettings.host}:${this.proxySettings.port}`;
+            agent = new HttpsProxyAgent(proxyUrl);
```

Comment:
I had a terrible idea, but could be something, but dont know how to do it, maybe pinging the proxy with `https`, if it is ok use that, otherwise `http`.

I dont know if this idea is necessary, maybe its handled inside the `HttpsProxyAgent` but couldnt tell from the information the library has.

Reply:
I know that HttpsProxyAgent supports both HTTP and HTTPS requests. First, it tries with one and then the other. 
But, I'm not sure if the proxy URL should be with the right protocol or if it doesn't matter.
I'll take a look on how to ping a proxy server URL, in case this solution doesn't work for the customer.